src/providers/delimitedtext/qgsdelimitedtextsourceselect.cpp

```diff
@@ -792,3 +839,153 @@ void QgsDelimitedTextSourceSelect::showCrsWidget()
+      query.addQueryItem( QStringLiteral( "yField" ), field );
+    }
+    if ( !cmbZField->currentText().isEmpty() )
+    {
+      field = cmbZField->currentText();
+      query.addQueryItem( QStringLiteral( "zField" ), field );
+    }
+    if ( !cmbMField->currentText().isEmpty() )
+    {
+      field = cmbMField->currentText();
+      query.addQueryItem( QStringLiteral( "mField" ), field );
+    }
+  }
+  else if ( geomTypeWKT->isChecked() )
+  {
+    if ( ! cmbWktField->currentText().isEmpty() )
+    {
+      const QString field = cmbWktField->currentText();
+      query.addQueryItem( QStringLiteral( "wktField" ), field );
+    }
+    if ( cmbGeometryType->currentIndex() > 0 )
+    {
+      query.addQueryItem( QStringLiteral( "geomType" ), cmbGeometryType->currentText() );
+    }
+  }
+  else
+  {
+    haveGeom = false;
+    query.addQueryItem( QStringLiteral( "geomType" ), QStringLiteral( "none" ) );
+  }
+  if ( haveGeom )
+  {
+    const QgsCoordinateReferenceSystem crs = crsGeometry->crs();
+    if ( crs.isValid() )
+    {
+      query.addQueryItem( QStringLiteral( "crs" ), crs.authid() );
+    }
+
+  }
+
+  if ( ! geomTypeNone->isChecked() )
+  {
+    query.addQueryItem( QStringLiteral( "spatialIndex" ), cbxSpatialIndex->isChecked() ? QStringLiteral( "yes" ) : QStringLiteral( "no" ) );
+  }
+
+  query.addQueryItem( QStringLiteral( "subsetIndex" ), cbxSubsetIndex->isChecked() ? QStringLiteral( "yes" ) : QStringLiteral( "no" ) );
+  query.addQueryItem( QStringLiteral( "watchFile" ), cbxWatchFile->isChecked() ? QStringLiteral( "yes" ) : QStringLiteral( "no" ) );
+
+  if ( ! skipOverriddenTypes )
+  {
+    // Set field types if overridden
+    for ( int column = 0; column < tblSample->columnCount(); column++ )
+    {
+      const QString fieldName { tblSample->horizontalHeaderItem( column )->text() };
+      const int fieldIdx { mFields.lookupField( fieldName ) };
+      if ( fieldIdx >= 0 )
+      {
+        QComboBox *typeCombo { qobject_cast<QComboBox *>( tblSample->cellWidget( 0, column ) ) };
+        const QString fieldTypeName { mFields.field( fieldName ).typeName() };
+        if ( typeCombo && typeCombo->currentData().toString() != fieldTypeName )
+        {
+          QgsDebugMsgLevel( QStringLiteral( "Overriding field %1 from %2 to %3" ).arg( fieldName, fieldTypeName, typeCombo->currentData().toString() ), 2 );
+          query.addQueryItem( QStringLiteral( "field" ),
+                              QString( fieldName ).replace( ':', QStringLiteral( "%3A" ) ) + ':' +  typeCombo->currentData().toString() );
+        }
+      }
+    }
+  }
+
+  url.setQuery( query );
+  return QString::fromLatin1( url.toEncoded() );
+}
+
+void QgsDelimitedTextSourceSelect::cancelScanTask()
+{
+  // This will cancel the existing task (if any)
+  if ( mScanTaskId > 0 )
+  {
+    QgsDelimitedTextFileScanTask *task { qobject_cast<QgsDelimitedTextFileScanTask *>( QgsApplication::taskManager()->task( mScanTaskId ) ) };
```

Comment:
Could you store the current task in a QPointer member instead and directly access here?